index.src.html

```diff
@@ -661,22 +671,30 @@ <h3 id="queue-report" algorithm>
       :   [=report/attempts=]
       ::  0
 
-  2.  If |url| was not provided by the caller, let |url| be |settings|'s
+  2.  Let |environment| be |settings|'s <a>realm execution context</a>'s
+      <a spec=ecmascript>realm</a>'s <a spec=ecmascript lt=realm>ECMAScript
+      global environment</a>.
+
+  3.  Execute [[#notify-observers]] with |environment| and |report|.
+
+  4.  If |settings|'s <a>report count</a> contains an entry
+      with <a spec=infra>key</a> |type|, then increment that entry. Otherwise,
+      create an entry in |settings|'s <a>report count</a>
+      with <a spec=infra>key</a> |type| and <a spec=infra>value</a> 1.
+
+  5.  If |settings|'s <a>report count</a>[|type|] is greater than 100 and the
+      current <a spec="netinfo">saveData</a> preference is true, return.
```

Comment:
@dcreager curious, do you have a sense for the rough expected size of each report upload today? I'd love to understand the overall cost of this feature to the user. 

@paulmeyer90 I think what you're proposing is reasonable, but I believe it diverges from our current implementation in Chrome — correct? As in, we have a hard cap of 100 reports (per type) regardless of data saver preference.

Reply:
For the current Chrome implementation, there's at most 1 upload per minute for each URL in the reporting policy.  The upload would contain a batch of reports; the size of the upload would depend on the number of reports generated during that minute.

For NEL, I don't have access to the sizing data for Google's real-world reports anymore, but @sburnett might.  The example reports in the spec clock in at 300-400 bytes each.  The biggest variable is the URL; everything else is a pretty consistent size.  (Note that's just for NEL; other types of report might be much larger and/or more variable in size.)

Reply:
Currently in Chrome, there is a 100 report limit in the report buffer (we won't buffer more than that, so if many more come at once, still only 100 will be delivered), but no overall limit for the number of reports delivered. It also currently does not behave differently when data saver is on.

Reply:
> For the current Chrome implementation, there's at most 1 upload per minute for each URL in the reporting policy.

@dcreager that's a great point. We don't currently spec this.. should we? This seems like a good pattern that we should encourage; I've heard many past complaints about CSP violations triggering dozens to hundreds of requests. 

> Currently in Chrome, there is a 100 report limit in the report buffer (we won't buffer more than that, so if many more come at once, still only 100 will be delivered), but no overall limit for the number of reports delivered.

@paulmeyer90 to confirm, the combined interaction here is: there is a single upload once a minute and the buffer for an individual entry type is allowed to accumulate up to 100 reports within this window — is that correct?